R/slurmUtils.R

```diff
@@ -0,0 +1,249 @@
+                          paste0("hddcdd_", fileRow$gcm, "_", fileRow$rcp,
+                                 "_", fileRow$start, "-", fileRow$end, ".csv"))
+
+  # job name
+  jobName <- paste0(config$jobNamePrefix, "_", fileRow$gcm, "_", fileRow$rcp,
+                    "_", fileRow$start, "-", fileRow$end)
+
+  # write slurm job script
+  jobScript <- tempfile(pattern = "initcalc_job_", fileext = ".slurm")
+
+  writeLines(c(
+    "#!/bin/bash",
+    sprintf("#SBATCH --job-name=%s", jobName),
+    sprintf("#SBATCH --output=%s/%s.out", config$logsDir, jobName),
+    sprintf("#SBATCH --error=%s/%s.err", config$logsDir, jobName),
+    sprintf("#SBATCH --cpus-per-task=%s", config$cpusPerTask),
+    sprintf("#SBATCH --nodes=%s", config$nodes),
+    sprintf("#SBATCH --ntasks=%s", config$ntasks),
+    sprintf("#SBATCH --partition=%s", config$partition),
+    sprintf("#SBATCH --qos=%s", config$qos),
+    "",
+    "source /p/system/modulefiles/defaults/piam/module_load_piam",
+    "",
+    "R --no-save <<EOF",
+    "library(devtools)",
+    sprintf("devtools::load_all(\"/p/tmp/hagento/dev/climbed\")"),   # not sure how to properly manage this yet
+    "",
+    "# Load and restore raster with names",
+    sprintf("pop <- terra::rast('%s/pop_%s.nc')", tmpDir, batch_tag),
+    sprintf("pop_names <- readRDS('%s/pop_names_%s.rds')", tmpDir, batch_tag),
+    "names(pop) <- pop_names",
+    "",
+    sprintf("tLim <- readRDS('%s/tLim_%s.rds')", tmpDir, batch_tag),
+    sprintf("hddcddFactor <- readRDS('%s/hddcddFactor_%s.rds')", tmpDir, batch_tag),
+    sprintf("wBAIT <- readRDS('%s/wBAIT_%s.rds')", tmpDir, batch_tag),
+    "",
+    sprintf("fileMapping <- data.frame(
+      gcm = '%s',
+      rcp = '%s',
+      start = %s,
+      end = %s,
+      tas = '%s',
+      rsds = '%s',
+      sfcwind = '%s',
+      huss = '%s',
+      stringsAsFactors = FALSE)",
+      fileRow$gcm, fileRow$rcp, fileRow$start, fileRow$end,
+      fileRow$tas, fileRow$rsds, fileRow$sfcwind, fileRow$huss
+    ),
+    "",
+    "result <- initCalculation(",
+    "  fileMapping = fileMapping,",
+    sprintf("  ssp = '%s',", ssp),
+    sprintf("  bait = '%s',", bait),
+    "  tLim = tLim,",
+    "  pop = pop,",
+    "  hddcddFactor = hddcddFactor,",
+    "  wBAIT = wBAIT",
+    ")",
+    "",
+    sprintf("write.csv(result, '%s', row.names = FALSE)", outputFile),
+    "",
+    "# Clean up all temporary files",
+    sprintf("unlink(list.files('%s', pattern='%s', full.names=TRUE))", tmpDir, batch_tag),
+    "EOF"
+  ), jobScript)
+
+
+  # submit slurm job
+  slurmCommand <- sprintf("sbatch %s", jobScript)
+  submissionResult <- system(slurmCommand, intern = TRUE)
+  jobId <- str_extract(submissionResult, "\\d+")
+
+
+  # return relevant job information
+  return(list(jobName = jobName,
+              jobScript = jobScript,
+              outputFile = outputFile,
+              slurmCommand = slurmCommand,
+              jobId = jobId,
+              batch_tag = batch_tag))
```

Comment:
Make the input invisible such that it doesn't flush the console if the user did no assignment.
```suggestion
  return(invisible(list(jobName = jobName,
                        jobScript = jobScript,
                        outputFile = outputFile,
                        slurmCommand = slurmCommand,
                        jobId = jobId,
                        batch_tag = batch_tag)))
```